src/test/java/reposense/commits/CommitInfoAnalyzerTest.java

```diff
@@ -285,6 +288,38 @@ public void analyzeCommits_commitsWithBinaryFileContribution_success() throws Ex
         Assert.assertEquals(expectedCommitResults, actualCommitResults);
     }
 
+    @Test
+    public void analyzeCommits_fileNameWithSpecialChars_success() throws Exception {
```

Comment:
will be running this test on non-windows as the file paths contain invalid characters in Windows.

Note that in running main program, GitLsTree will first check if system is windows, and will validate the file paths before calling the `CommitInfoAnalyzer#analyzeCommits` method. So in Windows, those file paths with invalid windows characters will not reach this point.